rfc3164/machine.go.rl

```diff
@@ -43,6 +43,9 @@ action set_timestamp {
 		fhold;
 		fgoto fail;
 	} else {
+		if m.timezone != nil {
+			t, _ = time.ParseInLocation(time.Stamp, string(m.text()), m.timezone)
+		}
```

Comment:
Would you be open to a refactor here so that the time isn't parsed twice here where the first result ends up being ignored? 

I think it can be achieved by always using `ParseInLocation` but then with UTC if no custom option is given?

Reply:
Additionally, there's no release yet with this functionality included, would it be possible to make a new release as well (maybe even after adding this optimization?).

Reply:
I'd love to look at a PR with such optimization @dbussink

Reply:
```
diff --git a/rfc3164/machine.go.rl b/rfc3164/machine.go.rl
index 4d13817..6629696 100644
--- a/rfc3164/machine.go.rl
+++ b/rfc3164/machine.go.rl
@@ -38,20 +38,37 @@ action set_prival {
 }

 action set_timestamp {
-       if t, e := time.Parse(time.Stamp, string(m.text())); e != nil {
-               m.err = fmt.Errorf("%s [col %d]", e, m.p)
-               fhold;
-               fgoto fail;
-       } else {
-               if m.timezone != nil {
-                       t, _ = time.ParseInLocation(time.Stamp, string(m.text()), m.timezone)
-               }
-               output.timestamp = t.AddDate(m.yyyy, 0, 0)
-               if m.loc != nil {
-                       output.timestamp = output.timestamp.In(m.loc)
-               }
-               output.timestampSet = true
-       }
+       if m.timezone != nil {
+        if t, e := time.ParseInLocation(time.Stamp, string(m.text()), m.timezone); e != nil {
+            m.err = fmt.Errorf("%s [col %d]", e, m.p)
+            (m.p)--
+
+            {
+                goto st373
+            }
+        } else {
+            output.timestamp = t.AddDate(m.yyyy, 0, 0)
+            if m.loc != nil {
+                output.timestamp = output.timestamp.In(m.loc)
+            }
+            output.timestampSet = true
+        }
+    } else {
+        if t, e := time.Parse(time.Stamp, string(m.text())); e != nil {
+            m.err = fmt.Errorf("%s [col %d]", e, m.p)
+            (m.p)--
+
+            {
+                goto st373
+            }
+        } else {
+            output.timestamp = t.AddDate(m.yyyy, 0, 0)
+            if m.loc != nil {
+                output.timestamp = output.timestamp.In(m.loc)
+            }
+            output.timestampSet = true
+        }
+    }
 }

 action set_rfc3339 {
```
How about this one?